ssl/t1_lib.c

```diff
@@ -1062,6 +1062,48 @@ int tls_check_sigalg_curve(const SSL *s, int curve)
 }
 #endif
 
+/*
+ * Return the number of security bits for the signature algorithm, or 0 on
+ * error.
+ */
+static int sigalg_security_bits(const SIGALG_LOOKUP *lu)
+{
+    const EVP_MD *md = NULL;
+    int secbits = 0;
+
+    if (!tls1_lookup_md(lu, &md))
+        return 0;
+    if (md != NULL)
+    {
+        int md_type = EVP_MD_type(md);
+
+        /* Security bits: half digest bits */
+        secbits = EVP_MD_size(md) * 4;
+	/*
+	 * SHA1 and MD5 are known broken. Reduce security bits so that they're
+	 * no longer accepted at security level 1. The real values don't
+	 * really matter as long as they're lower than 80, which is our
+	 * security level 1.
+	 * https://eprint.iacr.org/2020/014 puts a chosen-prefix attack for
+	 * SHA1 at 2^63.4 and MD5+SHA1 at 2^67.2
+	 * https://documents.epfl.ch/users/l/le/lenstra/public/papers/lat.pdf
+	 * puts puts a chosen-prefix attack for MD5 at 2^39.
```

Comment:
nit : comment block indentation is is of by 2.
typo : repeated `puts`

Reply:
Somehow I ended up with tabs, while vim should be set up to never use tabs.